packages/next/types/index.d.ts

```diff
@@ -318,6 +318,21 @@ declare global {
     randomUUID(): string
   }
 
+  // TODO: remove this polyfill when it is adopted into the spec.
+  interface PromiseConstructor {
+    /**
+     * Creates a new promise with exposed resolvers to resolve/reject. This will
+     * be adopted into the spec as `Promise.withResolvers`.
+     *
+     * @see https://tc39.es/proposal-promise-with-resolvers/
+     */
+    withResolvers<T>(): {
+      promise: Promise<T>
+      resolve: (value: T | PromiseLike<T>) => void
+      reject: (reason?: any) => void
```

Comment:
Should this be `unknown` instead of `any`?

```suggestion
      reject: (reason?: unknown) => void
```